app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -44,8 +43,8 @@ def mint_doi(work)
         work.update(doi: [register_doi.identifier])
       end
 
-      def can_mint_for?(work)
-        work.doi.present? || work.visibility != "open" || workflow_state(work)&.workflow_state_name != "deposited"
+      def cant_mint_for?(work)
```

Comment:
It is better to use positive names for methods returning truth values to avoid double negation.  Why not use !can_mint_for(work) and then define that as:
````
def can_mint_for(work)
  work.creator.present? || work.doi.blank? || work.visibility == "open" || workflow_state(work)&.workflow_state_name == "deposited"
end

Reply:
Thanks just seeing this.

Reply:
@thenapking which do you say is preferred? because you said positive names is preferred  for method returning truth  but proposed the proposed name **!cant_mint_for(work)** looks like  the negative version instead of the current positive sounding version. 

Did i get you wrong.

Reply:
is an if check not to run if any of those is true, but this returns false:

         false unless work.creator.present? 
         
  While in **mint_doi** method we have 
  
       return if cant_mint_for?(work)

Reply:
I can switch that to unless

Reply:
I would do this if you are short of time:
````
def can_mint_for(work)
  !(work.creator.blank? || work.doi.present? || work.visibility != "open" || workflow_state(work)&.workflow_state_name != "deposited")
end
````
And then call `!can_mint`.  It would then be worth refactoring this code as soon as you do have time.

Reply:
Thanks I will go with that with a small change instead of calling **!cant_mint** another negation, I will just switch **if** to **unless**

Reply:
I don't know what the conditions really are here and it's difficult to know whether you are changing things to correct a typo or to clear up the logic.

For me the change will be to correct typo from cant to can.